plugins/bazaar/src/components/DeleteProjectDialog/DeleteProjectDialog.tsx

```diff
@@ -0,0 +1,236 @@
+        </IconButton>
+      ) : null}
+    </MuiDialogTitle>
+  );
+});
+
+const DialogContent = withStyles((theme: Theme) => ({
+  root: {
+    padding: theme.spacing(2),
+  },
+}))(MuiDialogContent);
+
+const DialogActions = withStyles((theme: Theme) => ({
+  root: {
+    margin: 0,
+    padding: theme.spacing(1),
+  },
+}))(MuiDialogActions);
+
+type Props = {
+  entity: Entity;
+  openDelete: boolean;
+  handleClose: () => void;
+};
+
+export const DeleteProjectDialog = ({
+  entity,
+  openDelete,
+  handleClose,
+}: Props) => {
+  const auth = useApi(githubAuthApiRef);
+
+  const { value } = useAsync(async (): Promise<string[]> => {
+    return await getBranches(auth, entity);
+  }, []);
+
+  const branch = useRef(value?.[0] || '');
+  const [, setBranchState] = useState(branch.current);
+
+  const [title, setTitle] = useState('Delete project from the Bazaar');
+  const [commitMessage, setCommitMessage] = useState(
+    'update catalog-info.yaml',
+  );
+  const isInvalid = useRef(false);
+  const [isFormInvalid, setIsFormInvalid] = useState(false);
+
+  const handleTitleChange = (event: ChangeEvent<HTMLInputElement>): void => {
+    setTitle(event.target.value);
+  };
+
+  const handleCommitMessageChange = (
+    event: ChangeEvent<HTMLInputElement>,
+  ): void => {
+    setCommitMessage(event.target.value);
+  };
+
+  const handleBranchChange = (branchName: string) => {
+    branch.current = branchName;
+    setBranchState(branchName);
+  };
+
+  const clearForm = () => {
+    setTitle('Delete project from the Bazaar');
+    setCommitMessage('update catalog-info.yaml');
+    setIsFormInvalid(false);
+    isInvalid.current = false;
+    setBranchState('');
+  };
+
+  const handleCloseAndClear = () => {
+    handleClose();
+    clearForm();
+  };
+
+  const validate = () => {
+    if (commitMessage === '' || title === '' || branch.current === '') {
+      isInvalid.current = true;
+      setIsFormInvalid(true);
+    } else {
+      isInvalid.current = false;
+      setIsFormInvalid(false);
+    }
+  };
+
+  const handleSubmit = async () => {
+    validate();
+
+    if (!isInvalid.current) {
+      const clonedEntity = deleteProperties(JSON.parse(JSON.stringify(entity)));
+      delete clonedEntity.metadata.bazaar;
+
+      if (clonedEntity.metadata.tags === ['bazaar']) {
+        delete clonedEntity.metadata.tags;
+      } else {
+        clonedEntity.metadata.tags = clonedEntity.metadata.tags?.filter(
+          tag => tag !== 'bazaar',
+        );
+      }
+
+      await createPullRequest(
+        auth,
+        title,
+        commitMessage,
+        branch.current,
+        clonedEntity,
+      );
+      handleCloseAndClear();
+    }
+  };
+
+  return (
+    <Dialog
+      fullWidth
+      maxWidth="xs"
+      onClose={handleCloseAndClear}
+      aria-labelledby="customized-dialog-title"
+      open={openDelete}
+    >
+      <DialogTitle id="customized-dialog-title" onClose={handleCloseAndClear}>
+        Delete project
```

Comment:
Part of me wonders if this might be simpler to store the Bazaar information in a plugin-specific database rather than the entity descriptor files. This seems counter to Backstage's normal GitOps model, but Backstage doesn't offer entity CRUD -- this is generally expected to be handled outside of the system.

The add/edit/delete here are a bit of a strange hybrid where CRUD _appears_ to be possible -- but ultimately just spawns PRs. This comes at the expense of making it difficult to support multiple SCM providers, and needing to fetch branches on delete.

Reply:
The reason we decided to go with PRs to edit an entity is because when editing an entity in Backstage you are redirected to edit the catalog-info.yaml file. However, we could change the workflow if it's more intuitive to change the information right away and store all the Bazaar information together in a database. We would be happy to proceed with just the database if you can confirm that it is Backstage's preferred solution?

Reply:
We decided to try storing all the Bazaar information in a database instead. However, we encountered an issue in the function that decides if the Bazaar card should be displayed on the entity page (isBazaarAvailable). If we store the Bazaar information in the database we would need to fetch the projects in that function and compare them to the entity. But we cannot pass an async function to the EntitySwitch in EntityPage. We are thinking about if we should still add a Bazaar tag to the catalog-info.yaml, just to make it easier to check if a project belongs to the Bazaar, but that would still require a PR when adding the project to the Bazaar. What's your opinion on this?